### PR TITLE
Add Custom Timesteps Support to LCMScheduler and Supported Pipelines

### PR DESCRIPTION
**Pull request by dg845**
# What does this PR do?

This PR adds custom `timesteps` support to `LCMScheduler.set_timesteps` and pipelines which support `LCMScheduler` (e.g. `LatentConsistencyModelPipeline`, `StableDiffusionPipeline`, `StableDiffusionXLPipeline`). This gives users the flexibility to try out timestep schedules beyond the "standard" LCM inference schedule (as calculated when `num_inference_steps` is given to `LCMScheduler.set_timesteps`).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten
@patil-suraj
@luosiallen

### CODE REVIEW

File: src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py [hunk 1]
```diff
@@ -804,8 +809,19 @@ def __call__(
             prompt_embeds = torch.cat([negative_prompt_embeds, prompt_embeds])
 
         # 4. Prepare timesteps
-        self.scheduler.set_timesteps(num_inference_steps, device=device)
-        timesteps = self.scheduler.timesteps
+        if timesteps is not None:
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py:812]: very nice! 

Could we maybe factor this out into a function:

```py
timesteps = retrieve_timesteps(self.scheduler, timesteps)
```

and then use #copied from for all pipelines?

**Review by patrickvonplaten — COMMENTED** [review]
Thanks a lot for opening this PR - I think that's a good idea! Could we maybe try to make use of # Copied from by factoring out the function (similarly to: https://github.com/huggingface/diffusers/blob/6fac1369d0140ebeafdaeeef3f558f21fa3d5108/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py#L162)

**Comment by dg845** [comment]
@patrickvonplaten I have refactored the timestep handling logic into a `retrieve_timesteps` function located in `/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py` and copied it to all of the pipelines which support `LCMScheduler` with the `# Copied from` mechanism.

**Comment by patrickvonplaten** [comment]
> @patrickvonplaten I have refactored the timestep handling logic into a `retrieve_timesteps` function located in `/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py` and copied it to all of the pipelines which support `LCMScheduler` with the `# Copied from` mechanism.

This looks great - very nice job! Merging!

**Pull request merged**